test/module/irohad/model/converters/json_query_factory_test.cpp

```diff
@@ -181,4 +182,13 @@ TEST(QuerySerializerTest, get_role_permissions){
   auto val = queryGenerator.generateGetRolePermissions();
   val->signature = generateSignature(42);
   runQueryTest(val);
-}
\ No newline at end of file
+}
+
+TEST(QuerySerializerTest, get_account_asset_transactions) {
+  QueryGenerator queryGenerator;
+  auto val = queryGenerator.generateGetAccountAssetTransactions(
```

Comment:
Seems like val used only after dereference, maybe replace with `auto &val = *...`?

Reply:
Return value is rvalue and we cannot keep the content. So assigning the value is needed.

Reply:
I'll add:
```cpp
ASSERT_TRUE(val.has_value());
```

Reply:
Fixed it.